### PR TITLE
App engine configuration tests

### PR DESCRIPTION
**Pull request by ILMTitan**
* Split parts of AppEngineFlexDeployment into a separate service AppEngineConfiguration.
* Add tests for AppEngineConfiguration.
* Change GcpOutputWindow and ShellUtils to mockable services.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/Utils/IGcpOutputWindow.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace GoogleCloudExtension.Utils
+{
+    public interface IGcpOutputWindow
+    {
+        void Activate();
```
[the remark this answers is not in the document]
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension/Utils/IGcpOutputWindow.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/Services/Configuration/AppEngineConfigurationTests.cs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+        [TestMethod]
+        [DataRow(null)]
+        [DataRow("")]
+        [DataRow(AppEngineConfiguration.DefaultServiceName)]
+        public void TestSaveServiceToAppYaml_GeneratesNewDefaultFile(string defaultServiceName)
+        {
+            const string projectDirectory = @"c:\Project\Directory";
+            const string targetAppYaml = @"c:\Project\Directory\app.yaml";
+            _parsedProjectMock.Setup(p => p.DirectoryPath).Returns(projectDirectory);
```
[the remark this answers is not in the document]
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtensionUnitTests/Services/Configuration/AppEngineConfigurationTests.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs [hunk 3]
```diff
@@ -124,6 +127,16 @@ public sealed class GoogleCloudExtensionPackage : Package, IGoogleCloudExtension
         /// </summary>
         public string VersionedApplicationName => $"{ApplicationName}/{ApplicationVersion}";
 
+        /// <summary>
+        /// The defualt <see cref="IShellUtils"/> service.
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs]: nit: default

File: GoogleCloudExtension/GoogleCloudExtension/Services/Configuration/AppEngineConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+            else
+            {
+                return AspNetCoreRuntime;
+            }
+        }
+
+        /// <summary>
+        /// Creates an app.yaml in the target directory that targets the given service.
+        /// If the given project contains an app.yaml, it is the source of the new file.
+        /// Otherwise, a defualt app.yaml is created.
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension/Services/Configuration/AppEngineConfiguration.cs]: nit: default

File: GoogleCloudExtension/GoogleCloudExtension/Services/Configuration/IAppEngineConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        /// Updates or creates the app.yaml for the project to target the given service.
+        /// </summary>
+        /// <param name="project">The project to update the app.yaml of.</param>
+        /// <param name="service">The name of the service to target.</param>
+        void SaveServiceToAppYaml(IParsedDteProject project, string service);
+
+        /// <summary>
+        /// Creates an app.yaml in the target directory that targets the given service.
+        /// If the given project contains an app.yaml, it is the source of the new file.
+        /// Otherwise, a defualt app.yaml is created.
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension/Services/Configuration/IAppEngineConfiguration.cs]: nit: default

**Pull request merged**